tagstudio/src/qt/widgets/tag_box.py

```diff
@@ -73,13 +93,33 @@ def __init__(
             f"background: #555555;"
             f"}}"
         )
+
         tsp = TagSearchPanel(self.driver.lib)
-        tsp.tag_chosen.connect(lambda x: self.add_tag_callback(x))
+        tsp.tag_chosen.connect(
+            lambda x: (
+                self.add_tag_callback(x),
+                self.tag_entry.clear(),
+            )
+        )
         self.add_modal = PanelModal(tsp, title, "Add Tags")
-        self.add_button.clicked.connect(
-            lambda: (
-                tsp.update_tags(),
-                self.add_modal.show(),
+
+        self.add_button.clicked.connect(self.add_modal.show)
+        self.tag_entry.textChanged.connect(
+            lambda text: (
+                tsp.search_field.setText(text),
+                tsp.update_tags(text),
```

Comment:
I dont understand the logic here - you have initiated `TagCompleter`, which contains all tag names for the autocompleter. Why do you need also query the database each time when the input value is changed?

Reply:
It was because if it didn't update, the add tag modal wouldn't have the updated tags, but I can probably change it to update only when the modal opens to reduce calls to the database.